src/xcode/ENA/ENA/Source/Scenes/FileScanner/FileScannerCoordinatorViewModel.swift

```diff
@@ -27,46 +25,45 @@ class FileScannerCoordinatorViewModel: NSObject, PHPickerViewControllerDelegate,
 
 	@available(iOS 14, *)
 	func picker(_ picker: PHPickerViewController, didFinishPicking results: [PHPickerResult]) {
-		defer {
-			self.dismiss()
-		}
-
-		// each result represents a selected image
-		results.forEach { result in
-			let itemProvider = result.itemProvider
-			guard itemProvider.canLoadObject(ofClass: UIImage.self) else {
-				return
-			}
-			itemProvider.loadObject(ofClass: UIImage.self) { [weak self]  provider, _ in
-				guard let self = self,
-					  let image = provider as? UIImage,
-					  let codes = self.findQRCodes(in: image),
-					  !codes.isEmpty
-				else {
-					Log.debug("Looks like we have an issue reading the image", log: .fileScanner)
+		dismiss()
+		showHUD { [weak self] in
```

Comment:
Why does the processing of the results needs to wait for the hud to show up? There is no user interaction or some other async process in between. At least I don't see it. And all is happening on the main thread anyways.